pkg/cache/lrucache.go

```diff
@@ -49,16 +50,27 @@ type GitRepoLRUCache struct {
 
 // NewGitRepoLRUCache returns a new NewGitRepoLRUCache configured with the
 // destination directory to cache git repos and minimum free gbs
-//
-// TODO - this should probably error if the amount of disk given by minFreeGbs
-// exceeds the actual amount of disk available on the drive (which is a misconfiguration).
 func NewGitRepoLRUCache(dir string, minFreeGbs uint64) (*GitRepoLRUCache, error) {
 	path := filepath.Clean(dir)
 	_, err := os.Stat(path)
 	if err != nil {
 		return nil, fmt.Errorf("error checking provided cache directory: %s", err.Error())
 	}
 
+	stats := &syscall.Statfs_t{}
+
+	err = syscall.Statfs(dir, stats)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching stats for cache directory: %s", err.Error())	
+	} 
+
+	freeSpace := stats.Bavail * uint64(stats.Bfree)
```

Comment:
Whoops! This cast should be for `stats.Bsize`:

```go
freeSpace := stats.Bavail * uint64(stats.Bsize)
```

So we're essentially getting the number of available blocks * the size of blocks = the amount of free space in bytes.

Reply:
Ops! missed that